packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -885,6 +886,15 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     const newValue = newProps.value;
     workInProgress.memoizedProps = newProps;
 
+    if (__DEV__) {
+      checkPropTypes(
+        workInProgress.type.propTypes,
```

Comment:
Let's only call `checkPropTypes` if `workInProgress.type.propTypes` exists. Read it before the call and then, if it exists, call the function, passing it as an argument.